pkg/executor/show.go

```diff
@@ -1195,6 +1195,15 @@ func constructResultOfShowCreateTable(ctx sessionctx.Context, dbName *ast.CIStr,
 	publicIndices := make([]*model.IndexInfo, 0, len(tableInfo.Indices))
 	for _, idx := range tableInfo.Indices {
 		if idx.State == model.StatePublic {
+			softDeleteColumn := slices.ContainsFunc(idx.Columns, func(s *model.IndexColumn) bool {
+				return model.IsSoftDeleteColumn(s.Name)
+			})
+			activeActiveColumn := slices.ContainsFunc(idx.Columns, func(s *model.IndexColumn) bool {
+				return model.IsActiveActiveColumn(s.Name)
+			})
+			if softDeleteColumn || activeActiveColumn {
+				continue
+			}
```

Comment:
It's better to use the comment format to show the indices with some softdlete or active-active columns. But we can  just show these indices before comment format finished.